### PR TITLE
[Kotlin] Bootstrap concepts reference

### PR DESCRIPTION
**Pull request by dector**

### CODE REVIEW

File: languages/kotlin/reference/README.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+- [Variables](https://kotlinlang.org/docs/reference/basic-syntax.html#defining-variables) (Unpacked)
+- [Comments](https://kotlinlang.org/docs/reference/basic-syntax.html#comments) (Unpacked)
+- [String templates](https://kotlinlang.org/docs/reference/basic-syntax.html#using-string-templates) (Unpacked)
+- [Conditional expressions](https://kotlinlang.org/docs/reference/basic-syntax.html#using-conditional-expressions) (Unpacked)
+- [Nullable values and null checks](https://kotlinlang.org/docs/reference/basic-syntax.html#using-nullable-values-and-checking-for-null) (Unpacked)
+- [Type checks and automatic casts](https://kotlinlang.org/docs/reference/basic-syntax.html#using-type-checks-and-automatic-casts) (Unpacked)
+- [for loop](https://kotlinlang.org/docs/reference/basic-syntax.html#using-a-for-loop) (Unpacked)
+- [while loop](https://kotlinlang.org/docs/reference/basic-syntax.html#using-a-while-loop) (Unpacked)
+- [when expression](https://kotlinlang.org/docs/reference/basic-syntax.html#using-a-while-loop) (Unpacked)
+- [Ranges](https://kotlinlang.org/docs/reference/basic-syntax.html#using-a-while-loop) (Unpacked)
```
Comment (eparovyshnaya) [inline languages/kotlin/reference/README.md]: Looks like copy-paste mistake. Consider dedicated ref for *Ranges*: https://kotlinlang.org/docs/reference/basic-syntax.html#using-ranges

File: languages/kotlin/reference/README.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+- [Collections](https://kotlinlang.org/docs/reference/basic-syntax.html#using-collections) (Unpacked)
+- [Creating basic classes and their instances](https://kotlinlang.org/docs/reference/basic-syntax.html#using-collections) (Unpacked)
```
Comment (eparovyshnaya) [inline languages/kotlin/reference/README.md]: And here too. Should it be https://kotlinlang.org/docs/reference/basic-syntax.html#creating-basic-classes-and-their-instances ?

File: languages/kotlin/reference/README.md (hunk 1 above)
Comment (dector) [reply languages/kotlin/reference/README.md]: Indeed. :)

File: languages/kotlin/reference/README.md (hunk 2 above)
Comment (dector) [reply languages/kotlin/reference/README.md]: Yup. Will remove this section as it's only a high-level plan for the document.

**Comment by dector** [comment]
@exercism/kotlin Huge changes. Please, check.

**Pull request merged**